src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectAccessor.cs

```diff
@@ -105,6 +105,7 @@ public async Task OpenProjectXmlForWriteAsync(UnconfiguredProject project, Actio
             Requires.NotNull(project, nameof(project));
             Requires.NotNull(project, nameof(action));
 
+            await ((ConfiguredProject2)project).EnsureProjectEvaluatedAsync();
```

Comment:
Pardon my ignorance about this area, but why is it crucial for the project to have been evaluated at this point?

Reply:
In the cache mode, even the msbuild construction model is cached for imports and properties from the imports. The xml/construction model write would need working with actual construction xmls. Perhaps it would make sense to decouple need to load the actual construction model vs need for evaluation given a construction model write would trigger an evaluation anyway down the road.

Reply:
The more I think about this, I feel this should be removed. I am planning to switch out the construction model selectively out of the cache mode on demand instead of full cache mode break out for the project.